src/OpalCompiler-Core/OCRequestorScope.class.st

```diff
@@ -39,6 +39,9 @@ OCRequestorScope >> findVariable: aBlock ifNone: aNotFound [
 { #category : #lookup }
 OCRequestorScope >> lookupVar: name [
 
+	"do not create bindings if we lookup for declaration"
+	LookupForDeclaration value ifTrue: [ ^ outerScope lookupVar: name ].
```

Comment:
I don't understand how this avoids to create bindings. This is not clear in the code...